doc/guides/dts/howtos.rst

```diff
@@ -204,21 +204,35 @@ variable :makevar:`DTC_OVERLAY_FILE` contains a space- or semicolon-separated
 list of overlays. If :makevar:`DTC_OVERLAY_FILE` specifies multiple files, they
 are included in that order by the C preprocessor.
 
-Here are some ways to set it:
-
-1. on the cmake build command line
-   (``-DDTC_OVERLAY_FILE="file1.overlay;file2.overlay"``)
-#. with the CMake ``set()`` command in the application ``CMakeLists.txt``,
-   before including zephyr's :file:`boilerplate.cmake` file
-#. create a ``boards/<BOARD>_<revision>.overlay`` file in the application
-   folder for the current board revision. This requires that the board supports
+Devicetree overlays can be specified by the user directly, in a
+:file:`CMakeLists.txt` file, or applied automatically.
+
+If an overlay is specified by the user or directly in a :file:`CMakeLists.txt`
+file then no overlay files will be applied automatically.
+
+Overlay files are applied according to the following scheme:
+
+1. If ``DTC_OVERLAY_FILE`` is set, the overlay file(s) in it are merged and used
+   by the application. ``DTC_OVERLAY_FILE`` can be set in various ways:
+
+   1. In :file:`CMakeLists.txt`, before calling ``find_package(Zephyr)``
+
+   2. By passing ``-DDTC_OVERLAY_FILE="file1.overlay;file2.overlay;..."``,
+      either directly or via ``west``
+
+   3. From the CMake variable cache
```

Comment:
We must be explaining a hundred places how to set a CMake cache variable.

Could we move this to https://docs.zephyrproject.org/latest/guides/build/index.html?highlight=cmake#build-system-cmake and refer to it from here?

Reply:
> We must be explaining a hundred places how to set a CMake cache variable.
> 
> Could we move this to https://docs.zephyrproject.org/latest/guides/build/index.html?highlight=cmake#build-system-cmake and refer to it from here?

I see what you mean, but considering that if changing this into something like:
> By setting `DTC_OVERLAY_FILE="file1.overlay;file2.overlay;..."` in the CMake variable cache, see `<some-link>` for details.

compared to the current 3 lines, then I don't think we gain much.
We would have to:
- Describe the CMake variable to set, `DTC_OVERLAY_FILE`
- How the value is expected to look like `file1.overlay;file2.overlay;...`.
- Add a link.

but the reader reading about DTC overlay files and not knowing about CMake cache would now have to click the link, and then go back and continue reading about DTC overlay files.

A reader of this page is most likely new to Zephyr and maybe also CMake, so providing this info together makes it easier to read continuesly and not having to jump back anf forth.

So I'm not convinced that changing this would benefit the reader.

Reply:
I disagree.

We can cut down, and improve, on docs all over the place, not just here. And a user that does not intend to set DTC_OVERLAY_FILE does not need to skim generic CMake information.

Will leave this to your judgement as code owner for the build system.

Reply:
> Will leave this to your judgement as code owner for the build system.

yeah, but this is not really build system but documentation.
So would be nice with a tech-writer kind of person to assist, as how to provide the best user experience.

Reply:
I agree with @SebastianBoe that we should be linking to content that says once and for all how to do things. That's why I wrote https://github.com/zephyrproject-rtos/zephyr/commit/5edb6d5def80322759222d514418b84c00b8c7ed (and why I'm now fighting to bring it back in https://github.com/zephyrproject-rtos/zephyr/pull/40233).

Reply:
> I agree with @SebastianBoe that we should be linking to content that says once and for all how to do things. That's why I wrote [5edb6d5](https://github.com/zephyrproject-rtos/zephyr/commit/5edb6d5def80322759222d514418b84c00b8c7ed) (and why I'm now fighting to bring it back in #40233).

Thanks for the link.

I'm not against changing, but I think it then lacks a cleanup throughout our docset.

Just doing it for overlay files just makes the page standout from other places where CMake variable / environment variables are described, for example:
https://docs.zephyrproject.org/latest/guides/build/kconfig/setting.html#the-initial-configuration
https://docs.zephyrproject.org/latest/application/index.html#boards
https://docs.zephyrproject.org/latest/guides/modules.html#without-west

Reply:
>  That's why I wrote [5edb6d5](https://github.com/zephyrproject-rtos/zephyr/commit/5edb6d5def80322759222d514418b84c00b8c7ed) (and why I'm now fighting to bring it back in #40233).

I just took a second look at that PR #40233.
That PR describes environment settings, and nothing about `-D<variable>=<value>` so linking to that will not address the comment from @SebastianBoe
> We must be explaining a hundred places how to set a CMake cache variable.

cause it would still be required to state how to set a CMake variable.